### PR TITLE
Fixing typos in listeners.md

### PR DESCRIPTION
**Pull request by sebadur**
- ParserRuleContext does not have type parameters
- Confusing sentence cleaned up a little

### CODE REVIEW

File: doc/listeners.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ By default, ANTLR-generated parsers build a data structure called a parse tree o
...
 
-Because we specify phrase structure with a set of rules, parse tree subtree roots correspond to grammar rule names. ANTLR has a ParseTreeWalker that knows how to walk these parse trees and trigger events in listener implementation objects that you can create. ANTLR the tool generates listener interfaces for you also unless you, unless you turn that off with a commandline option. You can also have it generate visitors. For example from a Java.g4 grammar, ANTLR generates:
+Because we specify phrase structure with a set of rules, parse tree subtree roots correspond to grammar rule names. ANTLR has a ParseTreeWalker that knows how to walk these parse trees and trigger events in listener implementation objects that you can create. ANTLR generates listener interfaces for you also, unless you turn that off with a commandline option. You can also have it generate visitors. For example from a Java.g4 grammar, ANTLR generates:
```
Comment (sharwell) [inline doc/listeners.md]: :bulb: 

~~ANTLR the~~ **The ANTLR** tool generates

File: doc/listeners.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Assuming you've created a listener object called `MyListener`, here is how to ca
 JavaLexer lexer = new JavaLexer(input);
 CommonTokenStream tokens = new CommonTokenStream(lexer);
 JavaParser parser = new JavaParser(tokens);
-ParserRuleContext<Token> tree = parser.compilationUnit(); // parse
+ParserRuleContext tree = parser.compilationUnit(); // parse
```
Comment (sharwell) [inline doc/listeners.md]: :bulb:

```
JavaParser.CompilationUnitContext tree = ...
```

File: doc/listeners.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ Assuming you've created a listener object called `MyListener`, here is how to ca
...
  
 ParseTreeWalker walker = new ParseTreeWalker(); // create standard walker
```
Comment (sharwell) [inline doc/listeners.md]: :memo: This is actually supposed to be the following:

```java
ParseTreeWalker walker = ParseTreeWalker.DEFAULT;
```

**Comment by sharwell** [comment]
@parrt Are you requesting contributors.txt updates for documentation-only changes?

**Comment by parrt** [comment]
Yep, as they might contain code snippets.

**Comment by sebadur** [comment]
Signed & Updated due to sharwell's suggestions.

**Comment by sharwell** [comment]
LGTM :+1:

**Comment by parrt** [comment]
thank you!

**Comment by parrt** [comment]
and thanks to @sharwell for suggesting the move to repo for precisely this reason!

**Pull request merged**